### ISSUE TITLE
Convert reference and hypotheses to native schema

### ISSUE THREAD

**Issue by EyalLavi** [ebu/benchmarkstt#29]
In v1, the native schema includes words only:
```
[
 {text: "hello"},
 {text: "world"}
]
```
The transcripts returned from each vendors, and the reference, need to be converted to this native scheme before further processing.

**Comment by EyalLavi** [comment]
Include vendor name for easier downstream processing?

**Comment by EyalLavi** [comment]
Include vendor name for easier downstream processing?

**Comment by EyalLavi** [comment]
Decision in meeting today to not group WER by vendor in v1.

**Comment by EyalLavi** [comment]
Do we actually need this step for v1? The normalised reference is a simple text file. Hypotheses JSON files include the entire transcript as a single string that can be easily extracted. Normalisation and diff can both be performed on the text instead of a structured JSON. So do we have to go through this conversion at all if we're only comparing words?

**Comment by MikeSmithEU** [comment]
> So do we have to go through this conversion at all if we're only comparing words?

This step is already necessary imho, as we would end up using the standard format throughout the entire program, it would be double work to now build a text-based differ and later having to build a native schema differ as well (implementation wise this is a big difference). 
Also there can be different 'word segmentation' algorithms used (e.g. only split on spaces? also split on dashes?), which means that before analysis is done it needs to be pre-segmented into words in order to not be stuck to the default word segmentation from the differ algorithm.

**Comment by EyalLavi** [comment]
Summarising discussion with @MikeSmithEU. We will use JSON for the diff (although note that two identical JSON objects can be different textually, for example if one is nicely formatted). 

Deciding where to split words is linguistically complex and also depends on language and locality (consider `€1 234,56` vs `€1,234.56`). So we will not attempt to catch all characters that separate words. A word for the purpose of the schema will be any string between spaces/tab/CRLF. The assumption here is that the user will be responsible for creating normallization rules that minimize  the effects of this minimal logic (such as 'remove spaces and punctuations between two digits' for the example above).

However, for future proofing, it will be useful to preserve information lost in the process of converting to the native schema. There will be a JSON field that contains the pre-segmented text, but for the moment it will only be used internally and not be included in the output.

**Comment by EyalLavi** [comment]
Propose we change the field name `text` to `word` to make it more specific and to accommodate other types of text in future versions of the schema (like punctuation).

**Comment by MikeSmithEU** [comment]
> Propose we change the field name `text` to `word` to make it more specific and to accommodate other types of text in future versions of the schema (like punctuation).

My proposal is different: leave the field name `text`, but in the future we might add a `textType` field, which could specify eg. "word" (only current case) or "punctuation", this way we can group metrics by this type and provide eg. WER for words and "WER" for punctuation (PER, "Punctuation Error Rate"?).

**Comment by EyalLavi** [comment]
@MikeSmithEU agreed. AWS for example takes that approach and uses the field name "item", with "type": "pronunciation" | "punctuation" . Suggest we use "item" so the same schema could accommodate timed non-textual elements (e.g. sound recognition).

**Comment by MikeSmithEU** [comment]
> Suggest we use "item" so the same schema could accommodate timed non-textual elements (e.g. sound recognition).

Agreed with something like 'item' or 'contents', not really bothered about the actual name, an abstract name opens up possibilities without adding confusion like we would have with 'text' not really containing text.

**Comment by MikeSmithEU** [comment]
Example:

```
[
{"item": "hello", "type": "word"},
{"item": "!", "type": "punctuation"}
]
```

**Comment by EyalLavi** [comment]
Decision in meeting today: use `item` and `type`.

**Comment by MikeSmithEU** [comment]
This has been implemented in #48 

@EyalLavi I propose closing this issue and opening new issues for further enhancement of the Schema.

**Comment by EyalLavi** [comment]
Agreed. Closing.
